internal/cli/cmd/config/set.go

```diff
@@ -20,9 +20,9 @@ func newSetCommand() *cobra.Command {
 kargo config set project my-project
 `,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			cfg, err := config.LoadCLIConfig()
+			cfg, err := config.EnsureCLIConfig()
```

Comment:
This doesn't seem necessary. Can't we just let this fail if no config file exists? If no config file exists, the user is not logged in, so changing anything else in the configuration seems meaningless.

Reply:
> but I think our UX strategy for CLI is to have same interface (or behavior) with kubectl

I'm not against that, but we're currently not very similar to kubectl in terms of configuration. Our config is a single context only and tracks the server you're currently logged into.

By contrast, kubectl's supports many contexts and switching between contexts.

With kubectl, if you `kubectl config set ...`, that is applied to the current context, which is _not forgotten_ when you switch to some other context. With `kargo`, when you `kargo login` to a different server, all the old config is forgotten...

Which is what essentially makes it (currently) meaningless to set configuration values for a server you're not logged into.

I think we should have a conversation at some point about whether we want our CLI to support multiple contexts like kubectl does. I see reasons to... I also see reasons not to. To the extent that Kargo is supposed to be a single pane of glass over many clusters, it might not be common for users to interact with more than one Kargo.